pkgs/development/tools/misc/gdb/debug-info-from-env.patch

```diff
@@ -3,11 +3,13 @@ of directories with separate debugging information files.
 
 --- a/gdb/main.c
 +++ b/gdb/main.c
-@@ -551,3 +551,6 @@ captured_main_1 (struct captured_main_args *context)
+@@ -556,4 +556,8 @@ captured_main_1 (struct captured_main_args *context)
  
 -  debug_file_directory = relocate_gdb_directory (DEBUGDIR,
 +  debug_file_directory = getenv("NIX_DEBUG_INFO_DIRS");
 +
 +  if (debug_file_directory == NULL)
 +    debug_file_directory = relocate_gdb_directory (DEBUGDIR,
  						 DEBUGDIR_RELOCATABLE);
++  else debug_file_directory = xstrdup(debug_file_directory);
```

Comment:
Switching the order of branches (first xstrdup, then relocate) would improve readability; otherwise this LGTM. Thank you!

Reply:
Thanks for the review. I've updated with your suggested change.